students/TimPauley/session02/code/charges_calc_update.py

```diff
@@ -0,0 +1,157 @@
+1: Only error messages.
+2: Error messages and warnings.
+3: Error messages, warnings and debug messages.
+You need to implement three types of logging messages:
+Debug: General comments, indicating where in the script flow we are. Should be shown on screen only (i.e., never saved to logfile).
+Warning: Used for missing elements in the source data that force a change in the flow. Shown on screen and on the log file.
+Error: Used for inconsistencies in the source data that will cause the script to crash or report incorrect results. Shown on screen and on the log file.
+Use the following format for your log messages:
+
+log_format = “%(asctime)s %(filename)s:%(lineno)-3d %(levelname)s %(message)s”
+
+Use the following filename format to timestamp your log files:
+
+log_file = datetime.datetime.now().strftime(“%Y-%m-%d”)+’.log’
+'''
+
+'''
+Returns total price paid for individual rentals
+'''
+
+#pylint: disable=W0703
+
+import logging
+import argparse
+import json
+import datetime
+import math
+
+
+
+log_format = "%(asctime)s %(filename)s:%(lineno)-3d %(levelname)s %(message)s"
+formatter = logging.Formatter(log_format)
+logger = logging.getLogger()
+
+
+def parse_cmd_arguments():
+    """
+    Parses commands from the commandline
+    """
+    debug_logging_level_help = 'debug logging level \n\
+                        0: No debug messages or log file.\n\
+                        1: Only error messages.'
+
+    parser = argparse.ArgumentParser(description='Process some integers.')
+    parser.add_argument('-i', '--input', help='input JSON file', required=True)
+    parser.add_argument(
+        '-o', '--output', help='ouput JSON file', required=True)
+    parser.add_argument(
+        '-d', '--debug', help=debug_logging_level_help, required=False, default=0)
+    return parser.parse_args()
+
+
+def confgure_logging(debug_lever):
+    """"
+    configures the logging
+    """
+    format_logger = '%(asctime)s %(filename)s:%(lineno)-3d%(levelname)s'
+    format_logger += ' %(message)s'
+    log_file = datetime.datetime.now().strptime('%Y-%m-%d') + '.log'
+    debug_leve = int(debug_level)
+    if debug_level == 0:
+        logging.basicConfig(level=logging.CRITICAL)
+    if debug_level > 0:
+        logging.basicConfig(format=format_logger, level=logging.ERROR)
+        f_handler = logging.FileHandler(log_file)
+        f_handler.setLevel(logging.WARNING)
+        logging.getLogger().addHandler(f_handler)
+    if debug_level == 2:
+        logging.getLogger().setLevel(logging.WARNING)
+    if debug_level == 3:
+        logging.getLogger().setLevel(logging.DEBUG)        
+
+def load_rentals_file(filename):
+    """
+    Loads the json file of rental data
+    """
+    logging.debug("load_rentals_file")
+    logging.debug("Opens json file and readz it in")
+    with open(filename) as file:
+        try:
+            data = json.load(file)
+        except:
+            logging.error(error_msg)
+            logging.debug(i)
+    return data
```

Comment:
Remember if things go wrong, you should let your user know too, not just log it.